relentless/core.py

```diff
@@ -69,7 +69,12 @@ def __init__(self, x, y):
         if not np.all(x[1:] > x[:-1]):
             raise ValueError('x must be strictly increasing')
         self._domain = (x[0],x[-1])
-        self._spline = scipy.interpolate.Akima1DInterpolator(x=x, y=y)
+        if x.shape[0] > 2:
+            self._spline = scipy.interpolate.Akima1DInterpolator(x=x, y=y)
+        else:
+            self._spline = scipy.interpolate.interp1d(x=x, y=y)
+            self._y = y
+            self._x = x
```

Comment:
This works, but if you want a linear interpolation that supports the derivative method, you could use
```py
self._spline = scipy.interpolate.InterpolatedUnivariateSpline(x=x, y=y, k=1)
```
I think this class is available in most versions of scipy. If not, you can fall back on `UnivariateSpline` and additionally set `s=0`.